### PR TITLE
Update PMOD and USB timing constraints

### PR DESCRIPTION
**Pull request by elliotb-lowrisc**
1. Add constraints for the new PMOD C ports.
2. Update PMOD 0/1 constraints for SPI interface now present (along with I^2C and UART).
3. Relax USB I/O constraints a bit to aid overall timing.

### CODE REVIEW

**Review by GregAC — COMMENTED** [review]
LGTM thanks @elliotb-lowrisc. Of course we must check timing still looks good once rebased on the merged pinmux.

**Comment by GregAC** [comment]
Applying 1.0 label as good constraints are important for reliability. If we do start seeing timing issues I would suggest relaxing constraints (e.g. saying SPI can only run at half the speed it currently can) and documenting that we recommend slower speeds for these pins.

**Comment by elliotb-lowrisc** [comment]
Rebase

File: data/synth_timing.xdc [hunk 1]
```diff
@@ -292,20 +296,58 @@ set usb_conf_names {usrusb_softcn usrusb_sus usrusb_spd}
...
 set_output_delay -clock vclk_extusb -min 0 [get_ports $usb_conf_names]
 
-## PMOD 0 - asynchronous GPIO
+## PMOD 0
+#   SPI (or I^2C or UART or GPIO, but let us assume SPI is most critical)
+# Use *similar* methodology as Arduino Shield SPI, but with less I/O delay on
+# some shared ports to avoid timing violations on I^2C and UART paths.
+# Distance to PMOD 0 SPI pins = ~20-25 mm  x2(clk there + data back)
+set pmod0_spi_trce_dly_min [expr {20 * 2 * 0.004}]
+set_input_delay -clock clk_pmod0_spi -max [expr {$sclk_ns - 22                         }] [get_ports {pmod0[2]}] ;# CIPO
```
Comment (marnovandermaas) [inline data/synth_timing.xdc:305]: Why is this minus 22. In the Arduino it is minus 16.

File: data/synth_timing.xdc [hunk 2]
```diff
@@ -292,20 +296,58 @@ set usb_conf_names {usrusb_softcn usrusb_sus usrusb_spd}
...
+set_input_delay -clock vclk_sys -max 0 [get_ports $pmod0_gpio_names]
+set_input_delay -clock vclk_sys -min 0 [get_ports $pmod0_gpio_names]
+set_output_delay -clock vclk_sys -max 0 [get_ports $pmod0_gpio_names]
+set_output_delay -clock vclk_sys -min 0 [get_ports $pmod0_gpio_names]
+
+## PMOD 1
+#   SPI (or I^2C or UART or GPIO, but let us assume SPI is most critical)
+# Use *similar* methodology as Arduino Shield SPI, but with less I/O delay on
+# some shared ports to avoid timing violations on I^2C and UART paths.
+# Distance to PMOD 1 SPI pins = ~25-30 mm  x2(clk there + data back)
```
Comment (marnovandermaas) [inline data/synth_timing.xdc:327]: Is this larger than the 20 mentioned above for pmod0?

File: data/synth_timing.xdc [hunk 3]
```diff
@@ -798,6 +850,18 @@ set_multicycle_path        $ah_spi_mulcycs       -setup -start -to [get_ports {a
 set_multicycle_path [expr {$ah_spi_mulcycs - 1}] -hold  -start -to [get_ports {ah_tmpio11 ah_tmpio10}] ;# out
 set_multicycle_path        $ah_spi_mulcycs       -setup -end -from [get_ports ah_tmpio12] ;# in
 set_multicycle_path [expr {$ah_spi_mulcycs - 1}] -hold  -end -from [get_ports ah_tmpio12] ;# in
+## PMOD 0 SPI - use same methodology as Arduino SPI
+set pmod0_spi_mulcycs $ah_spi_mulcycs
```
Comment (marnovandermaas) [inline data/synth_timing.xdc:854]: Nit: maybe ah_spi_mulcycs should be caused external header cycles.

**Review by marnovandermaas — COMMENTED** [review]
Just put a few comments in from a code review.

File: data/synth_timing.xdc (hunk 1 above)
Comment (elliotb-lowrisc) [reply data/synth_timing.xdc:305]: This is relaxed a bit compared to the Arduino SPI to make the I^2C paths happy. Supposedly documented in the comment above, though perhaps that isn't clear.

File: data/synth_timing.xdc (hunk 2 above)
Comment (elliotb-lowrisc) [reply data/synth_timing.xdc:327]: Ah, thanks for catching this. I updated the comment but not the code!

File: data/synth_timing.xdc (hunk 3 above)
Comment (elliotb-lowrisc) [reply data/synth_timing.xdc:854]: Could do, although I use the same 'name based on first block' approach for some other interfaces too. Would you like me to rename them as well?

Comment (elliotb-lowrisc) [reply data/synth_timing.xdc:854]: I might as well

**Comment by elliotb-lowrisc** [comment]
Address review

**Comment by elliotb-lowrisc** [comment]
Reversed mistake of adjusting the USB I/O delays the wrong way (classic phys mistake)

File: data/synth_timing.xdc (hunk 3 above)
Comment (marnovandermaas) [reply data/synth_timing.xdc]: Thanks

File: data/synth_timing.xdc (hunk 2 above)
Comment (marnovandermaas) [reply data/synth_timing.xdc:327]: Resolved now.

**Comment by GregAC** [comment]
@marnovandermaas are you happy for this to go in now? Keen to get this merged due to the timing pressure issues and we can always iterate on it.

**Review by marnovandermaas — APPROVED** [review]
Yes, I have confirmed that the build is now faster with the new relaxed timing constraints. I want to test that the USB device still works after this but I can do that after merging.

**Pull request merged**